adm_program/system/classes/Email.php

```diff
@@ -565,135 +577,137 @@ public function setTemplateText($text, $senderName, $senderEmail, $senderUuid, $
         $this->emHtmlText = $emailHtmlText;
     }
 
+
     /**
-     * Funktion um den Nachrichtentext an die Mail uebergeben
-     * @param string $text
+     * Add the user specific template text to the email message and replace the plaeholders of the template.
+     * @param string $text        Email text that should be send
+     * @param string $firstname  Receiver firstname
+     * @param string $surname Receiver surname
+     * @param string $email  Receiver email address
+     * @param string $name  Receiver firstname and surname
      */
-    public function setText($text)
+    public function setUserSpecificTemplateText($text, $firstName, $surname, $email, $name)
     {
         // replace all line feeds within the mailtext into simple breaks because only those are valid within mails
         $text = str_replace("\r\n", "\n", $text);
 
-        $this->emText .= strip_tags($text);
-        $this->emHtmlText .= $text;
+        // replace parameters in email template
+        $replaces = array(
+            '#receiver_first_name#' => $firstName,
+            '#receiver_firstname#'  => $firstName,
+            '#receiver_surname#'    => $surname,
+            '#receiver_lastname#'   => $surname,
+            '#receiver_email#'      => $email,
+            '#receiver_name#'       => $name
+        );
+        return StringUtils::strMultiReplace($text, $replaces);
     }
 
     /**
-     * Sends a copy of the mail back to the sender. If the flag emListRecipients it set than all
-     * recipients will be listed in the mail.
-     * @throws \PHPMailer\PHPMailer\Exception
+     * Funktion um den Nachrichtentext an die Mail uebergeben
+     * @param string $text
      */
-    private function sendCopyMail()
```

Comment:
Why did you remove the sendCopyMail method? As I see there is no alternativ implementation of this feature. Within the copy mail you got the information to which roles the mail was sent.